app/services/dual-output/dual-output.ts

```diff
@@ -360,80 +381,291 @@ export class DualOutputService extends PersistentStatefulService<IDualOutputServ
+  convertSingleOutputToDualOutputCollection() {
+    // if in dual output mode, convert the single output collection to dual output
+    this.scenesService.views.scenes.forEach(scene => {
+      this.createVerticalNodes(scene.id);
+    });
+  }
+
+  /**
+   * Create a vertical node to partner with the vertical node
+   * @param horizontalNode - Node to copy to the vertical display
+   *
+   * @remark The horizontal node id is always the key in the scene node map.
+   * The node map entry is so that the horizontal and vertical nodes can refer to each other.
+   */
+  createVerticalNode(horizontalNode: TSceneNode, verticalNodeId?: string): TSceneNode {
+    const scene = horizontalNode.getScene();
+
+    if (horizontalNode.isFolder()) {
+      // add folder and create node map entry
+      const folder = scene.createFolder(horizontalNode.name, {
+        id: verticalNodeId,
+        display: 'vertical',
+      });
+      this.sceneCollectionsService.createNodeMapEntry(scene.id, horizontalNode.id, folder.id);
+
+      this.sceneNodeHandled.next();
+      return folder;
+    } else {
+      // add item
+      const item = scene.addSource(horizontalNode.sourceId, {
+        id: verticalNodeId,
+        display: 'vertical',
+      });
+
+      // position all of the nodes in the upper left corner of the vertical display
+      // so that all of the sources are visible
+      item.setTransform({ position: { x: 0, y: 0 } });
+
+      // show all vertical scene items by default
+      item.setVisibility(true);
+
+      // match locked
+      item.setLocked(horizontalNode.locked);
+
+      this.sceneCollectionsService.createNodeMapEntry(scene.id, horizontalNode.id, item.id);
+
+      this.sceneNodeHandled.next();
+      return item;
     }
   }
 
   /**
-   * Assign or confirm node contexts to a dual output scene
-   * @param sceneId - Id of the scene to map
+   * Create a horizontal node to partner with the vertical node
+   * @param verticalNode - Node to copy to the horizontal display
+   *
+   * @remark The horizontal node id is always the key in the scene node map.
+   * The node map entry is so that the horizontal and vertical nodes can refer to each other.
    */
-  confirmOrAssignSceneNodes(sceneId: string) {
-    this.SET_IS_LOADING(true);
-    const sceneItems = this.scenesService.views.getSceneItemsBySceneId(sceneId);
-    if (!sceneItems) return;
+  createHorizontalNode(
+    verticalNode: TSceneNode,
+    repair: boolean = false,
+    horizontalNodeId?: string,
+  ): TSceneNode {
+    const scene = verticalNode.getScene();
+
+    if (verticalNode.isFolder()) {
+      // add folder and create node map entry
+      const folder = scene.createFolder(verticalNode.name, {
+        id: horizontalNodeId,
+        display: 'horizontal',
+      });
+      folder.placeBefore(verticalNode.id);
+
+      this.sceneCollectionsService.createNodeMapEntry(scene.id, folder.id, verticalNode.id);
+
+      folder.placeBefore(verticalNode.id);
```

Comment:
I don't know is `placeBefore` is idempotent or not, but it looks like we're calling it twice? We're not doing the same for `item` below, so I wonder if there's something special with folders.

Reply:
Thanks for catching! Done